src/streaming/controllers/ScheduleController.js

```diff
@@ -203,47 +203,62 @@ function ScheduleController(config) {
         return request;
     }
 
-    function replaceCanceledRequests(canceledRequests) {
-        var ln = canceledRequests.length;
+    function replaceRequests(reqArr) {
         // EPSILON is used to avoid javascript floating point issue, e.g. if request.startTime = 19.2,
         // request.duration = 3.83, than request.startTime + request.startTime = 19.2 + 1.92 = 21.119999999999997
-        var EPSILON = 0.1;
-        var request,
-            time,
-            i;
+        let EPSILON = 0.1;
+        let request,
+            time;
 
-        for (i = 0; i < ln; i++) {
-            request = canceledRequests[i];
+        for (let i = 0, ln = reqArr.length; i < ln; i++) {
+            request = reqArr[i];
             time = request.startTime + (request.duration / 2) + EPSILON;
             request = adapter.getFragmentRequestForTime(streamProcessor, currentRepresentationInfo, time, {timeThreshold: 0, ignoreIsFinished: true});
```

Comment:
I know you didn't write it, but the reuse of request here is nasty. We should avoid doing this.

NIT: `request` and `time` could scoped in the for loop, in which case they could be `const`, subject to fixing above. `EPSILON` should be `const` rather than `let`.